ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala

```diff
@@ -173,3 +173,49 @@ final class CommandsValidator(ledgerId: LedgerId) {
         Left(missingField("command"))
     }
 }
+
+object CommandsValidator {
+
+  case class Submitters[T](actAs: Set[T], readAs: Set[T])
+
+  def effectiveSubmitters(commands: Option[ProtoCommands]): Submitters[String] = {
+    commands.fold(noSubmitters)(effectiveSubmitters)
+  }
+
+  def effectiveSubmitters(commands: ProtoCommands): Submitters[String] = {
+    val effectiveActAs =
+      if (commands.party.isEmpty)
+        commands.actAs.toSet
+      else
+        commands.actAs.toSet + commands.party
+    val effectiveReadAs = commands.readAs.toSet -- effectiveActAs
+    Submitters(effectiveActAs, effectiveReadAs)
+  }
+
+  val noSubmitters: Submitters[String] = Submitters(Set.empty, Set.empty)
+
+  def validateSubmitters(
+      commands: ProtoCommands): Either[StatusRuntimeException, Submitters[Ref.Party]] = {
+    def actAsMustNotBeEmpty(effectiveActAs: Set[Ref.Party]) =
+      if (effectiveActAs.isEmpty)
+        Left(missingField("party or act_as"))
+      else
+        Right(())
+
+    // Temporary check to reject all multi-party submissions until they are implemented
+    // Note: when removing this method, also remove the call to `actAs.head` in validateCommands()
+    def requireSingleSubmitter(actAs: Set[Ref.Party], readAs: Set[Ref.Party]) =
+      if ((actAs ++ readAs).size != 1)
```

Comment:
At this stage we already have a disjoint set of parties, so we could avoid joining the sets and getting the size of each set individually? It likely doesn't matter though.